relay-general/src/store/transactions/processor.rs

```diff
@@ -396,6 +407,45 @@ impl Processor for TransactionsProcessor<'_> {
         Ok(())
     }
 
+    /// Validates the length and characters of measurement names.
+    fn process_measurements(
+        &mut self,
+        value: &mut crate::protocol::Measurements,
+        _meta: &mut Meta,
+        _state: &ProcessingState<'_>,
+    ) -> ProcessingResult {
+        let measurements = &mut value.0;
+
+        measurements.retain(|key, val| {
+            if !is_valid_metric_name(key) {
+                relay_log::error!("Measurement name contains invalid characters");
```

Comment:
hmm how would it work since it's looping over a lot of values, do you mean return Err immediatly if a measurement is deleted or in the end if any invalid ones are found?

Reply:
This would log an error for _us_, but we actually want to inform the user about this. So best write to `Meta`, like we do here: https://github.com/getsentry/relay/blob/0575289402d1cac2c29485a7dee8fc151ce39014/relay-general/src/store/normalize.rs#L295-L306